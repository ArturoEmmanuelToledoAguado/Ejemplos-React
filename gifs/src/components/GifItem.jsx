export const GifItem = ({title,url}) => {

    console.log(url)

    const copyPath = ()=>{
        navigator.clipboard.writeText(url)
    }

  return (
    <div className="">
        <img 
            src={url}
            alt={title}
        />
        <p>{title}</p>
        <button className=""onClick={copyPath}>Copy Path</button>
    </div>
  )
}
