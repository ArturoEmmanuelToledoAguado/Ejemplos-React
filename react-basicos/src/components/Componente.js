import React from "react";
//import React,{Component} from "react";

const Componente=props=> <h2>{props.msg}</h2>

/* Componente en funcion declarada
function Componente(props) {
    return <h2>{props.msg}</h2>;
}*/

/* Componente con clase
class Componente extends Component{
    render() {
        return <h2>{this.props.msg}</h2>;
    }
}*/

export default Componente;