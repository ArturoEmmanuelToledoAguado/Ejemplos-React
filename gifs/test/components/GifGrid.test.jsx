import { render, screen } from "@testing-library/react"
import { GifGrid } from "../../src/components/GifGrid"
import { useFetchGifs } from "../../src/hooks/useFetchGifs"

jest.mock("../../src/hooks/useFetchGifs")//Emula el resultado de mandar a llamar una funci√≥n

describe('Pruebas para el componente  <GifGrid>', () => { 
    const category='Evangelion'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        useFetchGifs.mockReturnValue({
            images:[],
            isLoading:true
        })
        const {container} = render(<GifGrid category={category}/>)
        expect(container).toMatchSnapshot()
     })

    test('Esta prueba es para mostrar el Loading inicialmente', () => { 
        useFetchGifs.mockReturnValue({
            images:[],
            isLoading:true
        })
        render(<GifGrid category={category}/>)
        expect(screen.getAllByText('Loading...'))
        expect(screen.getAllByText(category))
     })

    test('Esta prueba es para mostrar las imagenes con el useFetchGifs', () => { 
        const gifs = [{
            id:'rtyj',
            title: 'ghj',
            url:'https://github.com/'
        },{
            id:'rtyj',
            title: 'ghj',
            url:'https://github.com/'
        },{
            id:'rtyj',
            title: 'ghj',
            url:'https://github.com/'
        }]

        useFetchGifs.mockReturnValue({
            images:gifs,
            isLoading:false
        })
        render(<GifGrid category={category}/>)
        expect(screen.getAllByRole('img').length).toBe(3)
     })
 })