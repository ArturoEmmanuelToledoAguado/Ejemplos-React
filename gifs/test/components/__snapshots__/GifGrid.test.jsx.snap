// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente  <GifGrid> Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <h2>
    Evangelion
  </h2>
  <h2>
    Loading...
  </h2>
</div>
`;
